frontend/src/lib/components/AddToDashboard/AddToDashboard.tsx

```diff
@@ -1,31 +1,24 @@
-import { useState } from 'react'
-import { AddToDashboardModal } from './AddToDashboardModal'
 import { InsightModel } from '~/types'
 import { dashboardsModel } from '~/models/dashboardsModel'
 import { useValues } from 'kea'
 import { LemonButton } from 'lib/lemon-ui/LemonButton'
 import { IconGauge, IconWithCount } from 'lib/lemon-ui/icons'
-import { NewDashboardModal } from 'scenes/dashboard/NewDashboardModal'
 
 interface SaveToDashboardProps {
     insight: Partial<InsightModel>
-    canEditInsight: boolean
+    setOpenModal: (open: boolean) => void
 }
 
-export function AddToDashboard({ insight, canEditInsight }: SaveToDashboardProps): JSX.Element {
-    const [openModal, setOpenModal] = useState<boolean>(false)
+export function AddToDashboard({ insight, setOpenModal }: SaveToDashboardProps): JSX.Element | null {
     const { rawDashboards } = useValues(dashboardsModel)
     const dashboards = insight.dashboard_tiles?.map((tile) => rawDashboards[tile.dashboard_id]).filter((d) => !!d) || []
 
+    if (dashboards.length === 0) {
+        return null
+    }
```

Comment:
There's still the "Add to a new dashboard" button, so I think it'd be better to always show this, with just an empty state

Reply:
This was actually hiding the button, when the insight was not yet added to a dashboard. Not when there are no dashboards. Adding to a dashboard was then only possible via the menu. Re-adding this button with the other proposed changes ("Add to dashboard" / "Manage dashboard"), as this makes more sense in a second look.